### PR TITLE
[ARTS-756] Services use the shared role client

### PR DESCRIPTION
**Pull request by anatbal**
## Description

- [x] Practitioner uses the shared role service client
- [x] Security uses the shared role service client
- [x] Run on AZ

### Changes
- [ARTS-756] - Services use the shared role client

### Checklist:

- [x] Branch name follows convention (feature/arts-#-short-name OR fix/arts-#-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

[ARTS-756]: https://ndph-arts.atlassian.net/browse/ARTS-756

### CODE REVIEW

**Review by eladiw — APPROVED** [review]

**Review by alexbaxter — DISMISSED** [review]
OK to merge - lot of small changes but not really anything substantive

File: init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/config/WebClientConfig.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import reactor.netty.http.client.HttpClient;
+import reactor.util.retry.Retry;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+@Configuration
+@ConfigurationProperties(prefix = "http-client")
+public class WebClientConfig {
```
Comment (shajidissa) [inline init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/config/WebClientConfig.java:22]: Any reason why this was moved to config package from configuration package ?

File: init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/PractitionerServiceTest.java [hunk 2]
```diff
@@ -14,13 +14,13 @@
...
 import org.springframework.test.util.ReflectionTestUtils;
 import org.springframework.web.reactive.function.client.WebClient;
-import uk.ac.ox.ndph.mts.init_service.configuration.WebClientConfig;
 import uk.ac.ox.ndph.mts.init_service.config.AzureTokenService;
 import uk.ac.ox.ndph.mts.init_service.exception.DependentServiceException;
 import uk.ac.ox.ndph.mts.init_service.model.IDResponse;
 import uk.ac.ox.ndph.mts.init_service.model.Practitioner;
 import uk.ac.ox.ndph.mts.init_service.model.PractitionerUserAccount;
 import uk.ac.ox.ndph.mts.init_service.model.RoleAssignment;
+import uk.ac.ox.ndph.mts.roleserviceclient.configuration.WebClientConfig;
```
Comment (shajidissa) [inline init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/PractitionerServiceTest.java:23]: Can we revert to using uk.ac.ox.ndph.mts.init_service.configuration.WebClientConfig, make changes to match init_service WebClientConfig to roleserviceclient WebClientConfig

File: init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/RoleServiceTest.java [hunk 3]
```diff
@@ -11,11 +11,11 @@
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 import org.springframework.web.reactive.function.client.WebClient;
-import uk.ac.ox.ndph.mts.init_service.configuration.WebClientConfig;
 import uk.ac.ox.ndph.mts.init_service.config.AzureTokenService;
 import uk.ac.ox.ndph.mts.init_service.exception.DependentServiceException;
 import uk.ac.ox.ndph.mts.init_service.model.Permission;
 import uk.ac.ox.ndph.mts.init_service.model.Role;
+import uk.ac.ox.ndph.mts.roleserviceclient.configuration.WebClientConfig;
```
Comment (shajidissa) [inline init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/RoleServiceTest.java:18]: Same as above

File: init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/SiteServiceTest.java [hunk 4]
```diff
@@ -12,11 +12,11 @@
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 import org.springframework.web.reactive.function.client.WebClient;
-import uk.ac.ox.ndph.mts.init_service.configuration.WebClientConfig;
 import uk.ac.ox.ndph.mts.init_service.config.AzureTokenService;
 import uk.ac.ox.ndph.mts.init_service.exception.DependentServiceException;
 import uk.ac.ox.ndph.mts.init_service.model.IDResponse;
 import uk.ac.ox.ndph.mts.init_service.model.Site;
+import uk.ac.ox.ndph.mts.roleserviceclient.configuration.WebClientConfig;
```
Comment (shajidissa) [inline init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/SiteServiceTest.java:19]: Same as above

**Review by shajidissa — CHANGES_REQUESTED** [review]

File: init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/config/WebClientConfig.java (hunk 1 above)
Comment (anatbal) [reply init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/config/WebClientConfig.java:22]: configuration package is going to be deleted in the next PR.

File: init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/PractitionerServiceTest.java (hunk 2 above)
Comment (anatbal) [reply init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/PractitionerServiceTest.java:23]: In the next PR  uk.ac.ox.ndph.mts.init_service.configuration.WebClientConfig is going to be removed any way...

File: init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/RoleServiceTest.java (hunk 3 above)
Comment (anatbal) [reply init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/RoleServiceTest.java:18]: In the next PR  uk.ac.ox.ndph.mts.init_service.configuration.WebClientConfig is going to be removed any way...

File: init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/SiteServiceTest.java (hunk 4 above)
Comment (anatbal) [reply init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/SiteServiceTest.java:19]: In the next PR  uk.ac.ox.ndph.mts.init_service.configuration.WebClientConfig is going to be removed any way...

File: init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/PractitionerServiceTest.java (hunk 2 above)
Comment (shajidissa) [reply init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/PractitionerServiceTest.java:23]: Ok that's fine.

File: init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/RoleServiceTest.java (hunk 3 above)
Comment (shajidissa) [reply init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/RoleServiceTest.java:18]: ok

File: init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/SiteServiceTest.java (hunk 4 above)
Comment (shajidissa) [reply init-service/src/test/java/uk/ac/ox/ndph/mts/init_service/service/SiteServiceTest.java:19]: ok

**Review by shajidissa — DISMISSED** [review]

**Review by eladiw — APPROVED** [review]

**Review by alexbaxter — APPROVED** [review]
OK to merge

**Pull request merged**